### PR TITLE
Fireworks integration

### PR DESCRIPTION
**Pull request by rjanardhan3**
Description - Integrates Fireworks within Langchain LLMs to allow users to use Fireworks models with Langchain, mainly for summarization.

Issue - Not applicable
Dependencies - None
Tag maintainer - @rlancemartin

### CODE REVIEW

File: libs/langchain/langchain/llms/fireworks.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+            )
+        llm_output = {"token_usage": token_usage, "model_id": self.model_id}
+        return LLMResult(generations=generations, llm_output=llm_output)
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "fireworks"
+
+class FireworksChat(BaseLLM):
```
Comment (baskaryan) [inline libs/langchain/langchain/llms/fireworks.py:173]: should this be implemented as a ChatModel instead of BaseLLM? (and live in langchain/chat_models)

Comment (rjanardhan3) [reply libs/langchain/langchain/llms/fireworks.py:173]: We implemented FireworksChat(similar to OpenAIChat). We are currently implementing a chat completion API, which we we will then use to create ChatFireworks(similar to ChatOpenAI) which will live in langchain/chat_models

Comment (baskaryan) [reply libs/langchain/langchain/llms/fireworks.py:173]: OpenAIChat is deprecated in favor of ChatOpenAI, recommend against using that as a template. if it's easier we could just take FireworksChat out for now and in sep pr add ChatFireworks (no need to block this PR)

File: libs/langchain/langchain/llms/fireworks.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+) -> None:
+    """Update token usage."""
+    _keys_to_use = keys.intersection(response)
+    for _key in _keys_to_use:
+        if _key not in token_usage:
+            token_usage[_key] = response["usage"][_key]
+        else:
+            token_usage[_key] += response["usage"][_key]
+
+def execute(prompt, model, api_key, max_tokens=256, temperature=0.0, top_p=1.0) -> Any:
```
Comment (rlancemartin) [inline libs/langchain/langchain/llms/fireworks.py:295]: Add type annotations, as noted in test failures:
```
langchain/llms/fireworks.py:295: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
Found 1 error in 1 file (checked [14](https://github.com/langchain-ai/langchain/actions/runs/5675105724/job/15389065289?pr=8322#step:7:15)41 source files)
```

**Review by rlancemartin — COMMENTED** [review]
Nice work! I can run the notebook w/ provided API key.

IMO, we don't need to show the summarization example in the notebook.

Just showing LLMChain works is sufficient. 

But, include a few simple sections:

1/ Calling a model

Find a model on the xxx. 

Specify `model_name/version`.

For example, here is llamav2:
 
2/ Streaming support?

3/ How can parameters be passed?

File: libs/langchain/langchain/llms/fireworks.py (hunk 1 above)
Comment (rjanardhan3) [reply libs/langchain/langchain/llms/fireworks.py:173]: Sounds good to me!

**Comment by rjanardhan3** [comment]
The PR is updated - we didn't enable streaming in this initial version and were planning on integrating that as well in a follow-up PR!

**Comment by rlancemartin** [comment]
> The PR is updated - we didn't enable streaming in this initial version and were planning on integrating that as well in a follow-up PR!
> 
> 

Please run "make format" to resolve Lint errors.

**Pull request merged**